### PR TITLE
Base layer toggle

### PR DESCRIPTION
**Pull request by chrisvanmook**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
     "uuid": "^8.1.0"
   },
   "peerDependencies": {
-    "@datapunt/asc-assets": "0.19.2-alpha.66",
-    "@datapunt/asc-ui": "0.19.2-alpha.66",
+    "@datapunt/asc-assets": "0.19.2-beta.5",
+    "@datapunt/asc-ui": "0.19.2-beta.5",
```
Comment (jonkoops) [inline package.json:15]: We can pull these ones to the stable version.

File: package.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
     ]
   },
   "devDependencies": {
-    "@datapunt/asc-assets": "^0.19.2-alpha.66",
+    "@datapunt/asc-assets": "^0.19.2-beta.5",
     "@datapunt/asc-ui": "^0.19.2-beta.5",
```
Comment (jonkoops) [inline package.json:55]: Why are these listed both as `dependencies` as `devDependencies`? I would assume that `dependencies` would be enough.

File: src/components/BaseLayerToggle.tsx [hunk 3]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
-    }) => `url(${BASE_LAYER_STYLE[type].background[1]}) no-repeat top left /
+    background: ${({ type }) => `url(${
+      BASE_LAYER_STYLE[type as BaseLayerType].background[1]
+    }) no-repeat top left /
       120px 44px`};
   }
 `
+const Menu = styled(ContextMenu)`
+  & > button {
+    height: 44px;
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:89]: Could we resolve this with `themeSpacing`?

File: src/components/BaseLayerToggle.tsx [hunk 4]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+const BaseLayerToggle: React.FC = () => {
+  const [toggleBaseLayerType, setToggleBaseLayerType] = useState(
+    BaseLayerType.Topo,
+  )
+  const [selectedLayer, setSelectedLayer] = useState({
+    [BaseLayerType.Aerial]: AERIAL_AMSTERDAM_LAYERS[0].urlTemplate,
+    [BaseLayerType.Topo]: DEFAULT_AMSTERDAM_LAYERS[0].urlTemplate,
+  })
+
+  const handleToggle = () => {
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:122]: Should this be wrapped in a `useCallback`?

File: src/components/BaseLayerToggle.tsx [hunk 5]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+    setToggleBaseLayerType(
+      toggleBaseLayerType === BaseLayerType.Aerial
+        ? BaseLayerType.Topo
+        : BaseLayerType.Aerial,
+    )
+  }
+
+  const handleChange = (
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:130]: Should this be wrapped in a `useCallback`?

File: src/components/BaseLayerToggle.tsx [hunk 6]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+      setSelectedLayer({
+        ...selectedLayer,
+        [toggleBaseLayerType]: layer.urlTemplate,
+      })
+    }
+  }
+
+  return (
+    <StyledButton>
+      {/*
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:148]: Do we need this TS ignore?

File: src/components/BaseLayerToggle.tsx [hunk 7]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+      // @ts-ignore */}
+      <ToggleButton
+        variant="blank"
+        onClick={handleToggle}
+        type={toggleBaseLayerType}
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:153]: This statement seems to cause problems as `type` is used internally somehere. If I change this to `layerType` all type errors seem to go away.

File: src/components/BaseLayerToggle.tsx [hunk 8]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+        {toggleBaseLayerType?.toString()}
+      </ToggleButton>
+      <Menu
+        data-test="context-menu"
+        title="Actiemenu"
+        arrowIcon={<Ellipsis />}
+        selectElementForTouchScreen={
+          <>
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label style={{ display: 'none' }} htmlFor="some-id">
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:164]: Perhaps we can make this a little more descriptive than `"some-id"`?

File: src/components/BaseLayerToggle.tsx [hunk 9]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+        onClick={handleToggle}
+        type={toggleBaseLayerType}
+      >
+        {toggleBaseLayerType?.toString()}
+      </ToggleButton>
+      <Menu
+        data-test="context-menu"
+        title="Actiemenu"
+        arrowIcon={<Ellipsis />}
+        selectElementForTouchScreen={
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:161]: This prop doesn't seem to work for Firefox on mobile. I am seeing the custom dropdown and the layout seems to clip a little.

image

File: src/components/BaseLayerToggle.tsx [hunk 10]
```diff
@@ -62,32 +78,134 @@ const InnerBackground = styled.span<InnerImageProps>`
...
+          <StyledContextMenuItem
+            key={id}
+            onClick={(e) => {
+              setSelectedLayer({
+                ...selectedLayer,
+                [toggleBaseLayerType]: urlTemplate,
+              })
+              e.currentTarget.blur()
+            }}
+            {...(urlTemplate === selectedLayer[toggleBaseLayerType]
```
Comment (jonkoops) [inline src/components/BaseLayerToggle.tsx:192]: Would it not be easier to use a tenary here to set the icon prop?

File: package.json (hunk 2 above)
Comment (chrisvanmook) [reply package.json:55]: because they are listed in peerDependencies, so they won't be installed

File: src/components/BaseLayerToggle.tsx (hunk 3 above)
Comment (chrisvanmook) [reply src/components/BaseLayerToggle.tsx:89]: I think we shouldn't use themeSpacing for dimensions, only paddings and margins

File: src/components/BaseLayerToggle.tsx (hunk 6 above)
Comment (chrisvanmook) [reply src/components/BaseLayerToggle.tsx:148]: unfortunately we do

File: src/components/BaseLayerToggle.tsx (hunk 7 above)
Comment (chrisvanmook) [reply src/components/BaseLayerToggle.tsx:153]: nice one!

File: src/components/BaseLayerToggle.tsx (hunk 9 above)
Comment (chrisvanmook) [reply src/components/BaseLayerToggle.tsx:161]: Mobile mode in FF doesn't simulate touchevent I guess? try on chrome or a real mobile device on FF. Will fix the layout tho

File: package.json (hunk 2 above)
Comment (jonkoops) [reply package.json]: Peculiar, but ok :man_shrugging:

File: src/components/BaseLayerToggle.tsx (hunk 3 above)
Comment (jonkoops) [reply src/components/BaseLayerToggle.tsx:92]: Fair point.

**Pull request merged**